src/ethereum/web3.js

```diff
@@ -25,17 +25,34 @@ window.addEventListener('load', async () => {
     )
 })
 
-if (typeof window !== 'undefined' && typeof window.web3 !== 'undefined') {
-  console.log('Using the web3 object of the window...')
-  web3 = new Web3(window.web3.currentProvider)
+if (typeof window !== "undefined" && typeof window.web3 !== "undefined") {
```

Comment:
What happens if `window.ethereum` is undefined ? . What is the interest of instantiating it with `undefined` ? Wouldn't it be better to display a message to install Metamask?